### PR TITLE
[FLINK-25636][network] Change some default config values of blocking shuffle for better usability

### PR DESCRIPTION
**Pull request by wsry**
## What is the purpose of the change

As discussed for FLIP-199 (https://cwiki.apache.org/confluence/display/FLINK/FLIP-199%3A+Change+some+default+config+values+of+blocking+shuffle+for+better+usability). This PR aims to change some default config values of blocking shuffle for better usability.

## Brief change log

  - Increase the default read buffer size of sort-shuffle to 64M.
  - Increase the default write buffer size of sort-shuffle to 16M.
  - Make sort-shuffle the default shuffle implementation for batch jobs.

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**yes** / no)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: docs/layouts/shortcodes/generated/all_taskmanager_network_section.html [hunk 1]
```diff
@@ -136,15 +136,15 @@
...
+            <td>Parallelism threshold to switch between sort-based blocking shuffle and hash-based blocking shuffle, which means for batch jobs of smaller parallelism, hash-shuffle will be used and for jobs of larger parallelism, sort-shuffle will be used. The default value 1 means that sort-shuffle is the default option. Note: For production usage, you may also need to enable data compression by setting 'taskmanager.network.blocking-shuffle.compression.enabled' to true and tune 'taskmanager.network.sort-shuffle.min-buffers' and 'taskmanager.memory.framework.off-heap.batch-shuffle.size' for better performance.</td>
```
Comment (gaoyunhaii) [inline docs/layouts/shortcodes/generated/all_taskmanager_network_section.html:147]: `for jobs of larger parallelism` -> `for jobs of larger or equal parallelism` ?

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 2]
```diff
@@ -68,11 +68,11 @@ Flink [DataStream API]({{< ref "docs/dev/datastream/execution_mode" >}}) 和 [Ta
...
 
 当使用sort blocking shuffle的时候有些配置需要适配:
 - [taskmanager.network.blocking-shuffle.compression.enabled]({{< ref "docs/deployment/config" >}}#taskmanager-network-blocking-shuffle-compression-enabled): 配置该选项以启用 shuffle data 压缩，大部分任务建议开启除非你的数据压缩比率比较低。
-- [taskmanager.network.sort-shuffle.min-parallelism]({{< ref "docs/deployment/config" >}}#taskmanager-network-sort-shuffle-min-parallelism): 根据下游任务的并行度配置该选项以启用 sort shuffle。如果并行度低于设置的值，则使用 `hash shuffle`，否则 `sort shuffle`。
+- [taskmanager.network.sort-shuffle.min-parallelism]({{< ref "docs/deployment/config" >}}#taskmanager-network-sort-shuffle-min-parallelism): 根据下游任务的并行度配置该选项以启用 sort shuffle。如果并行度低于设置的值，则使用 `hash shuffle`，否则 `sort shuffle`。对于 1.15 以下的版本，它的默认值是 `Integer.MAX_VALUE`，所以默认情况下总是会使用 `hash shuffle`。从 1.15 开始，它的默认值是 1, 所以着默认情况下总是会使用 `sort shuffle`。
```
Comment (gaoyunhaii) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:75]: `着默认情况` -> `默认情况`

**Review by gaoyunhaii — COMMENTED** [review]
Thanks @wsry for opening the PR! The current change looks in general good to me and I have to minor comments.

One point is that compared with https://cwiki.apache.org/confluence/display/FLINK/FLIP-199%3A+Change+some+default+config+values+of+blocking+shuffle+for+better+usability,  it seems we do not enable `taskmanager.network.blocking-shuffle.compression.enabled` in this PR ?

**Comment by wsry** [comment]
@gaoyunhaii Thanks for your review and feedbacks. I have updated the PR. The change for ```taskmanager.network.blocking-shuffle.compression.enabled``` is in another PR: https://github.com/apache/flink/pull/17814. Another contributor has contributed that, you may also take a look at that.

**Review by gaoyunhaii — APPROVED** [review]
Thanks @wsry a lot for the update! Both the two PRs looks good to me~

**Pull request closed without merging**